src/core/cmcd/cmcd_data_builder.ts

```diff
@@ -0,0 +1,602 @@
+      props.ot = "i";
+    }
+
+    let precizeBufferLengthMs;
+    if (
+      lastObservation !== undefined &&
+      (props.ot === "v" || props.ot === "a" || props.ot === "av")
+    ) {
+      const bufferedForType = lastObservation.buffered[content.adaptation.type];
+      if (!isNullOrUndefined(bufferedForType)) {
+        // TODO more precize position estimate?
+        const position =
+          this._playbackObserver?.getCurrentTime() ??
+          lastObservation.position.getWanted() ??
+          lastObservation.position.getPolled();
+        for (const range of bufferedForType) {
+          if (position >= range.start && position < range.end) {
+            precizeBufferLengthMs = (range.end - position) * 1000;
+            props.bl = Math.floor(Math.round(precizeBufferLengthMs / 100) * 100);
+            break;
+          }
+        }
+      }
+    }
+
+    const precizeDeadlineMs =
+      precizeBufferLengthMs === undefined || lastObservation === undefined
+        ? undefined
+        : precizeBufferLengthMs / lastObservation.speed;
+
+    props.dl =
+      precizeDeadlineMs === undefined
+        ? undefined
+        : Math.floor(Math.round(precizeDeadlineMs / 100) * 100);
+
+    if (precizeDeadlineMs !== undefined) {
+      // estimate the file size, in kilobits
+      const estimatedFileSizeKb =
+        (content.representation.bitrate * content.segment.duration) / 1000;
+      const wantedCeilBandwidthKbps = estimatedFileSizeKb / (precizeDeadlineMs / 1000);
+      props.rtp = Math.floor(
+        Math.round((wantedCeilBandwidthKbps * RTP_FACTOR) / 100) * 100,
+      );
+    }
+
+    switch (content.manifest.transport) {
+      case "dash":
+        props.sf = "d";
+        break;
+      case "smooth":
+        props.sf = "s";
+        break;
+      default:
+        props.sf = "o";
+        break;
+    }
+    props.st = content.manifest.isDynamic ? "l" : "v";
+    props.tb = content.adaptation.representations.reduce(
+      (acc: number | undefined, representation: IRepresentation) => {
+        if (
+          representation.isSupported !== true ||
+          representation.decipherable === false
+        ) {
+          return acc;
+        }
+        if (acc === undefined) {
+          return Math.round(representation.bitrate / 1000);
+        }
+        return Math.max(acc, Math.round(representation.bitrate / 1000));
+      },
+      undefined,
+    );
+    return this._producePayload(props);
+  }
+
+  private _producePayload(props: ICmcdProperties): ICmcdPayload {
+    let cmcdObjectValue = "";
+    let cmcdRequestValue = "";
+    let cmcdSessionValue = "";
+    let cmcdStatusValue = "";
+    let queryStringPayload = "";
+    if (props.br !== undefined) {
+      queryStringPayload += `br=${String(props.br)},`;
+      const toAdd = `br=${String(props.br)},`;
+      if (this._typePreference === TypePreference.Headers) {
+        cmcdObjectValue += toAdd;
+      } else {
+        queryStringPayload += toAdd;
+      }
```

Comment:
`br=${String(props.br)},` is going to be added 2 times to `queryStringPayload`

Reply:
:O you're right, forgot to remove the first line as I was adding headers handling

Reply:
Updated